public/app/core/components/TransformersUI/rowsToFields/rowsToFields.ts

```diff
@@ -0,0 +1,193 @@
+import { map } from 'rxjs/operators';
+import {
+  anyToNumber,
+  ArrayVector,
+  DataFrame,
+  DataTransformerID,
+  DataTransformerInfo,
+  Field,
+  FieldColorModeId,
+  FieldConfig,
+  FieldType,
+  getFieldDisplayName,
+  ReducerID,
+  ThresholdsMode,
+} from '@grafana/data';
+
+export interface RowToFieldsTransformOptions {
+  nameField?: string;
+  valueField?: string;
+  mappings?: RowToFieldsTransformMappings[];
+}
+
+export interface RowToFieldsTransformMappings {
+  fieldName: string;
+  reducerId?: ReducerID;
+  configProperty: string;
```

Comment:
Perhaps we could use [keyof FieldConfig] | string?   

For the array based properties (thresholds, etc) we will likely need some better strategy -- maybe threshold[0]?

Reply:
Well it's not always an exact FieldConfig key, for example threshold1 etc,  yea, maybe threshold[1] but think using only alpha numerical chars could be an advantage to make it easier for query language / syntaxes that might prohibit or require escaping brackets `[ ]`